webhooks/elbv2/targetgroupbinding_validator.go

```diff
@@ -77,8 +77,19 @@ func (v *targetGroupBindingValidator) ValidateDelete(ctx context.Context, obj ru
 }
 
 // checkRequiredFields will check required fields are not absent.
-func (v *targetGroupBindingValidator) checkRequiredFields(tgb *elbv2api.TargetGroupBinding) error {
+func (v *targetGroupBindingValidator) checkRequiredFields(ctx context.Context, tgb *elbv2api.TargetGroupBinding) error {
 	var absentRequiredFields []string
+	if tgb.Spec.TargetGroupARN == "" {
+		if tgb.Spec.TargetGroupName == "" {
+			absentRequiredFields = append(absentRequiredFields, "either TargetGroupARN or TargetGroupName")
+		} else if tgb.Spec.TargetGroupName != "" {
+			tgObj, err := v.getTargetGroupsByNameFromAWS(ctx, tgb.Spec.TargetGroupName)
+			if err != nil {
+				return errors.Errorf("Can't locate TargetGroup with name %s", tgb.Spec.TargetGroupName)
+			}
+			tgb.Spec.TargetGroupARN = *tgObj.TargetGroupArn
```

Comment:
What is the purpose of this code? The validator can't mutate.

Reply:
The purpose of this code is to guarantee that the either the ARN of the TargetGroup exists or it's possible to infer the ARN by the name of the TargetGroup (since it's unique).

And even though the validator can't mutate, I added `tgb.Spec.TargetGroupARN = *tgObj.TargetGroupArn` to guarantee the object is in a consistent state though the rest of the process.

The whole code of `aws-load-balancer-controller` was written assuming there is an ARN. By changing the object here I guarantee as early as possible that that assumption is true.

Reply:
A comment to that effect would be helpful.

Reply:
Done